src/main/resources/lib/ssb/keyFigure.ts

```diff
@@ -86,7 +91,34 @@ export function parseKeyFigure(req: Request, keyFigure: Content<KeyFigure>, muni
   }
 
   const dataQueryId: string | undefined = keyFigure.data.dataquery
-  if (dataQueryId) {
+  const datasetRepo: DatasetRepoNode<JSONstat> | null = getDataset(keyFigure)
+
+  if (datasetRepo) {
+    const dataSource: KeyFigure['dataSource'] | undefined = keyFigure.data.dataSource
+    let data: JSDataset | Array<JSDataset> | null | TbmlData = datasetRepo.data
```

Comment:
Could we abstract out handling of jsonstat into a separate function and eliminate `let` ?

Reply:
Fixed